rero_mef/authorities/api.py

```diff
@@ -87,13 +87,121 @@ class MefRecord(AuthRecord):
     fetcher = mef_id_fetcher
     provider = MefProvider
 
+    @classmethod
+    def build_ref_string(cls, agency_pid, agency):
+        """Build ref string."""
+        from flask import current_app
+        with current_app.app_context():
+            url = current_app.config.get('JSONSCHEMAS_HOST')
+            ref_string = 'http://' + url + '/api/' + agency + '/' + agency_pid
+            return ref_string
+
+    @classmethod
+    def get_mef_by_agency_pid(cls, agency_pid, agency, with_deleted=False):
+        """Get mef record by agency pid value."""
+        key = agency + '.identifier_for_person'
+        search = MefSearch()
+        result = (
+            search.filter('term', **{key: agency_pid})
+            .source()
+            .execute()
+            .to_dict()
+        )
+        try:
+            mef_pid = result['hits']['hits'][0]['_source']['pid']
+            return cls.get_record_by_pid(mef_pid)
+        except IndexError:
+            return None
+
+    @classmethod
+    def get_mef_by_viaf_pid(cls, viaf_pid, with_deleted=False):
+        """Get mef record by agency pid value."""
+        search = MefSearch()
+        result = (
+            search.filter('term', viaf=viaf_pid).source().execute().to_dict()
+        )
+        try:
+            mef_pid = result['hits']['hits'][0]['_source']['pid']
+            return cls.get_record_by_pid(mef_pid)
+        except IndexError:
+            return None
+
+    @classmethod
+    def mef_sync(
```

Comment:
in other place the name was something like `create_or_update`...

Reply:
Fixed